File: lesson4/c_c++/homework/homework4_15.c
//: lesson4/c_c++/homework/homework4_15.c
#include <stdio.h>

int main() {
    int entered_numb;
    printf("Введите натуральное число => ");
    scanf("%d", &entered_numb);

    if (entered_numb > 0 && entered_numb < 100) {
        int result1 = entered_numb % 10;
        int result2 = entered_numb / 10 % 10;
        if (result1 == result2) {
            printf("Оба числа одинаковы.\n");
        } else {
            printf("Числа разнные.\n");
        }
    } else if (entered_numb > 100 && entered_numb < 1000) {
        int result1 = entered_numb / 10 % 10;
        int result2 = entered_numb / 100;
        if (result1 == result2) {
            printf("Первые две цыфры одинаковы.\n");
        } else {
            printf("Первые две цыфры неодинаковы.\n");
        }
    } else {
        int result1 = entered_numb / 100 % 10;
        int result2 = entered_numb / 1000;
        if (result1 == result2) {
            printf("Первые две цыфры одиноковы.\n");
        } else {
            printf("Первые две цыфры неодинаковы.\n");
        }
    }
    return 0;
}

File: lesson4/python3/homework/homework4_15.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_15.py
# Напишите программу, которая определяет, верно ли, что введенное число содержит
# две одинаковые цифры, стоящие рядом (221 или 122).

entered_numb = int(input("Введите натуральное число => "))

# numb_i = 0
if entered_numb > 0 and entered_numb < 100:
    result1 = entered_numb % 10
    result2 = entered_numb // 10 % 10
    if result1 == result2:
        print("Оба числа одинаковы.")
    else:
        print("Числа не одинаковы.")
elif entered_numb > 100 and entered_numb < 1000:
    result1 = entered_numb // 10 % 10
    result2 = entered_numb // 100
    if result1 == result2:
        print("Первые числа совпадают друг с другом.")
    else:
        print("Первые числа не совпадают друг с другом.")
else:
    result1 = entered_numb // 100 % 10
    result2 = entered_numb // 1000
    if result1 == result2:
        print("Первые числа совпадают друг с другом.")
    else:
        print("Первые числа не совпадают друг с другом.")

